packages/hint/src/lib/types/html.ts

```diff
@@ -74,13 +76,15 @@ export class HTMLElement {
          * need to restore it before return the outerHTML.
          */
         const fragment = htmlparser2Adapter.createDocumentFragment();
-        const parent = this._element.parent;
+        const { parent, next, prev } = this._element;
 
         htmlparser2Adapter.appendChild(fragment, this._element);
 
         const result = parse5.serialize(fragment, { treeAdapter: htmlparser2Adapter });
 
         this._element.parent = parent;
+        this._element.next = next;
+        this._element.prev = prev;
```

Comment:
For completeness you'll also want to set `next.prev = this._element` and `prev.next = this._element` (after checking to ensure they're not `null` of course).

Reply:
done!